components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -147,6 +162,51 @@ class SystemEngineView @JvmOverloads constructor(
         }
     }
 
+    internal fun handleLongClick(type: Int, extra: String): Boolean {
+        return when (type) {
+            EMAIL_TYPE -> {
```

Comment:
nit: We could simplify this by creating a `mapOf(EMAIL_TYPE to HitResult.EMAIL, GEO_TYPE to HitResult.GEO.....)`. Then we can handle all these cases at once with a map lookup.